src/analyses/region.ml

```diff
@@ -27,7 +27,7 @@ struct
       let ev = Reg.eval_exp exp in
       let to_exp (v,f) = (v,Lval.Fields.to_offs' f) in
       List.map to_exp (Reg.related_globals ev (part,reg))
-    | `Top -> Messages.warn "Region state is broken :("; []
+    | `Top -> Messages.warn (Messages.LogEvent.may (Messages.EventType.Unknown ("Region state is broken :("))); []
```

Comment:
This seems too verbose. Not specific to this, but in general.
Now with the non-polymorphic variants, there's no way around the path, but maybe change all those calls to 
`Messages.(warn (LogEvent.may (EventType.Unknown ("Region state is broken :("))))`?

Reply:
I think in most modules `Messages` was imported as just `M`. Would maybe some similar shortcuts help with the verbosity too? Or bring some more of the types a bit higher like `M.warn_may` and `M.warn_must` or `may_unknown`, `must_integer` and `may_integer` ?

Reply:
Yes. You can also just define aliases instead of pulling them out. Or as I wrote above, just put all types in one module. If there's only going to be the type and `show` for it, modules for each just makes it unnecessarily verbose (I assume they're not going to be put in some functor).

> I think in most modules Messages was imported as just M.

I meant the repeated `M.x M.y M.z` instead of `M.(x y z)`.

Reply:
Another way to make it less verbose would be to have the certainty as an optional argument for `warn` and `warn_each` as it is already done with `?ctx`:
`M.(warn Integer "...")`
`M.(warn ~must:true Integer "...")`
I assume we produce more `may` than `must`.

Reply:
The optional `must` parameter is now there. I also added some nesting and helper methods to build the types from the bottom so now it should be possible to do:

```
M.warn (M.Warning.Integer.overflow ());
M.warn (M.Warning.Behavior.Undefined.nullpointer_dereference ());
M.warn ~must:true (M.Warning.Behavior.Undefined.ArrayOutOfBounds.before_start ());
```

also the variants are now not hidden inside modules so the top level variants can be used directly as:

```
M.warn (M.Unknown "my message");
```

Reply:
@vandah Previously, there was the `Certainty.t option`, which would allow for `None` in case of "Debug" warnings. How is this handled now?

Reply:
This remained there. The `debug` function still works the same.

Before, I just called `warn` from the `debug` function. Now there's a function `warn_internal` which does what `warn` did previously and is called from both `warn` and `debug`. This `warn_internal` takes the type `WarningWithCertainty.t` which still has the `Certainty.t option` inside.